lib/src/revset.rs

```diff
@@ -2158,18 +2158,21 @@ impl VisibilityResolutionContext<'_> {
     }
 }
 
+pub trait RevsetIterator<T>: Iterator<Item = Result<T, RevsetEvaluationError>> {}
+impl<T, I: Iterator<Item = Result<T, RevsetEvaluationError>>> RevsetIterator<T> for I {}
```

Comment:
nit: it no longer make much sense to define new `RevsetIterator` trait. Just inline `Item = Result<..>`?